frontend/src/common/utils/checkForRevisionChange.ts

```diff
@@ -0,0 +1,121 @@
+import isEmpty from "lodash/isEmpty";
+import isEqual from "lodash/isEqual";
+import xorWith from "lodash/xorWith";
+import { Collection, Dataset } from "../entities";
+
+const IGNORED_COLLECTION_FIELDS = [
+  "visibility",
+  "created_at",
+  "updated_at",
+  "is_revision",
+  "revision_diff",
+  "datasets",
```

Comment:
probably need more context here, but I was imagining we'd need to detect change when someone updates a dataset with a new dropbox link or something?

Reply:
We explicitly check datasets and other arrays/objects